.github/workflows/pulumi-deploy.yml

```
@@ -0,0 +1,42 @@
+    paths:
+      - "pulumi/**"
+
+jobs:
+  deploy:
+    name: Deploy Infrastructure
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.8)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)